internal/venice-client-common/src/main/java/com/linkedin/venice/schema/rmd/RmdUtils.java

```diff
@@ -87,13 +87,13 @@ public static long sumOffsetVector(Object offsetVector) {
    * @return                True if the advancedOffset vector has grown beyond the baseOffset
    */
   static public boolean hasOffsetAdvanced(@NotNull List<Long> baseOffset, @NotNull List<Long> advancedOffset) {
```

Comment:
Not related to this code line here, but seems we did not take care of RMD offset vector for separate topic. As the offset vector's element is for each region, then we probably messed together normal real-time upstream offset and separate real-time upstream offset. This will cause trouble for Deterministic Conflict Resolution Validation, as it reads differet region's VT to detect violation by comparing real-time upstream offset. We may need a way to resolve this.

Reply:
Hmm, I am actually a bit confused by this. As I check the code in updateReplicationCheckpointVector method, which is used when doing all the AA operation, it will update the offset vector by kafkaClusterId, this ID is coming all the way from the StorePartitionDataReceiver. 
In the AggKafkaConsumerService#subscribeConsumerFor, when we are wiring the data receiver, we use the original KafkaURL and ID, which is unique for sep topic.

Reply:
For exmaple, if we have 3 regions for AA ingestion, without separate real-time topic, the offset vector will have 3 elements at most. If separate real-time topic is enabled, we will have 3 * 2 = 6 elements for offset vector at most?

Reply:
Yes, I updated the integration test to verify this. Please check TestSeparateRealtimeTopicIngestion#testIncrementalPushPartialUpdate. If I set source.grid.fabric to dc-1, it will have 4 value. If I don't set it (default to dc-0) it will be 3 value.